File: Semestres/2.Agosto-Diciembre 2017/FADA/Pila.cpp

// Example program
#include <iostream>
#include <string>
#include "Pila.h"
using namespace std;


Pila::Pila(){
	top = 0;
	for(int i=0; i<6; i++){
		arreglo[i]=0;    
	}
}

void Pila::Push(int x){
	top = top+1;
	if(top<=6){
		arreglo[top-1]=x;    
		
	}
	else{
		cout << "Overflow\n";    
	}
		
}

int Pila::Pop(){
	
	if(top==0){
		cout << "Underflow\n";
		return 0;
	}else{
		int elemento = arreglo[top-1];
		top-=1;
		return elemento;
	}
	
		
}

bool Pila::EmptyStack(){
	
	if(top==0) return true;
	return false;
}





int main()
{
    Pila *objPila = new Pila();
    cout << objPila->EmptyStack() << endl;
    objPila->Push(2);
    objPila->Push(5);
    cout << objPila->EmptyStack() << endl;
    cout << objPila->Pop() << endl;
    cout << objPila->EmptyStack() << endl;


}

File: Semestres/10.Agosto-Diciembre 2021/FADAUSB/16Oct/Pila.py
class Pila:
    def __init__(self,n):
        self.size = n
        self.pila = [None]*n
        self.top = 0

    def emptyStack(self):
        if self.top == 0:
            return True
        else:
            return False

    def push(self,x):
        if self.top == self.size:
            return "Overflow"
        else:
            self.top += 1
            self.pila[self.top-1] = x
            

    def pop(self):
        if self.top == 0:
            return "Underflow"
        else:
            a = self.pila[self.top-1]
            self.pila[self.top-1] = None
            self.top -= 1
            return a

"""
pila = Pila(5)
print(pila.pila)   
pila.push(5)
print(pila.pila)   
pila.push(4)
print(pila.pila)   
pila.push(3)
print(pila.pila)
print(pila.pop())
print(pila.pila)   
print(pila.pop()) 
print(pila.pila)        
pila.push(3)
pila.push(40)
print(pila.pila)
"""